Mathlib/MeasureTheory/Measure/Tight.lean

```diff
@@ -0,0 +1,469 @@
+/-
+Copyright (c) 2024 Josha Dekker. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rémy Degenne, Josha Dekker
+-/
+import Mathlib.MeasureTheory.Measure.Regular
+import Mathlib.Topology.Metrizable.Uniformity
+import Mathlib.Analysis.SpecificLimits.Basic
+
+/-!
+# (Pre-)tight measures
+The key definition of interest here is that of tight measures, `IsTight`. We first introduce two
+weaker notions, `IsSeparable` and `IsPretight`, which are equivalent on complete metric spaces. We
+provide some basic API for these notions and prove Ulam's tightness theorem
+(`of_isSeparableSpace_completeSpace`) and its strengthening `of_isSeparable_completeSpace`.
+
+## Main definitions
+* `IsSeparable`: A measure `μ` is separable if there is a separable set `S` such that
+  `μ S = μ Set.univ`.
+* `IsPretight`: A measure `μ` is pre-tight if for all `0 < ε`, there exists `K` totally bounded such
+  that `μ Kᶜ ≤ ε`.
+* `IsTight`: A measure `μ` is tight if for all `0 < ε`, there exists `K` compact such that
+  `μ Kᶜ ≤ ε`. This is defined in terms of filters. `IsTight.iff_compact_sets` establishes
+  equivalence with the usual definition.
+
+## Main statements
+
+* `of_isSeparableSpace_completeSpace`: Ulam's tightness theorem: a finite measure on a complete
+  separable metric space is tight.
+* `of_isSeparable_completeSpace`: A strengthening of Ulam's tightness theorem: a finite,
+  separable measure on a complete metric space is tight.
+* `of_innerRegular`: Every finite, inner-regular measure on a T2 space is tight.
+
+## Notation
+
+
+## Implementation details
+
+
+-/
+open Topology Filter Uniformity Uniform MeasureTheory Set
+open scoped ENNReal
+
+namespace MeasureTheory
+
+variable {α ι : Type*}
+
+variable [MeasurableSpace α] {μ : Measure α}
+
+/-- For a finite measure `μ`, we can extract from a countable cover that has full measure, a finite
+cover of accumulated sets that has `ε`-almost full measure. -/
+private lemma almost_cover_has_approx_accumulate [MeasurableSpace α] {μ : Measure α}
```

Comment:
This lemma shouldn't be private, as it could very well be useful in another context. Its name should rather be something like `exists_sub_le_measure_accumulate`, to follow more closely the symbols in the statement and make it more easily findable (your name mentions `almost`, `cover`, `approx` but none of these could be guessed by an educated reader).

Could you please review the other lemma names below with the same state of mind?

Reply:
Do you feel the same about the other lemma's I had made private before? I've removed `private` from them for now, although I'm not entirely sure what the nicest place for them would be. Should I run `find_home` on them, or should I keep them here, close to where they are actually used?